cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -540,31 +542,50 @@ private class CampaignManagerImpl(
         }
 
     override val hasLocalData: Boolean
-        get() = dataStorage.gdprConsentStatus != null || dataStorage.usPrivacyString != null
-
-    override fun handleMetaDataLogic(md: MetaDataResp?) {
-        metaDataResp = md
-        md?.let {
-            it.ccpa?.apply {
-                applies?.let { i -> dataStorage.ccpaApplies = i }
-                sampleRate?.let { i ->
-                    if (i != dataStorage.ccpaSamplingValue) {
-                        dataStorage.ccpaSamplingValue = i
-                        dataStorage.ccpaSamplingResult = null
-                    }
+        get() = dataStorage.gdprConsentStatus != null || dataStorage.uspstring.isNullOrEmpty().not()
+
+    override fun handleMetaDataResponse(response: MetaDataResp?) {
+        // update meta data response in the data storage
+        metaDataResp = response
+
+        if (response == null) return
+
+        // handle ccpa
+        response.ccpa?.apply {
+            applies?.let { i ->
+                ccpaConsentStatus?.let { ccpaCS ->
+                    val updatedCcpaConsentStatus = ccpaCS.copy(applies = i)
+                    dataStorage.ccpaConsentStatus =
+                        JsonConverter.converter.encodeToString(updatedCcpaConsentStatus)
                 }
             }
-            it.gdpr?.apply {
-                applies?.let { i -> dataStorage.gdprApplies = i }
-                childPmId?.let { i -> dataStorage.gdprChildPmId = i }
-                sampleRate?.let { i ->
-                    if (i != dataStorage.gdprSamplingValue) {
-                        dataStorage.gdprSamplingValue = i
-                        dataStorage.gdprSamplingResult = null
-                    }
+
+            sampleRate?.let { i ->
+                if (i != dataStorage.ccpaSamplingValue) {
+                    dataStorage.ccpaSamplingValue = i
+                    dataStorage.ccpaSamplingResult = null
+                }
+            }
+        }
+
+        // handle gdpr
+        response.gdpr?.apply {
+            applies?.let { i -> dataStorage.gdprApplies = i }
```

Comment:
I thought we'd get rid of `dataStorage.gdprApplies` and deal only with the consent object `.applies` attribute.

Reply:
@andresilveirah since this is a PR to fix CCPA related issue, I decided to refactor CCPA related code only, and for the GDPR code I can create a ticket to do it separatelly, since:
1. it will help us to merge this task asap
2. it will prevent us from making mistakes in the part of the code, that is not related to the issue we are trying to fix
3. it will prevent us from breaking the purpose of the pull request (which is to fix a problem with uspstring)